src/ucp/core/ucp_ep.c

```diff
@@ -623,6 +627,12 @@ void ucp_ep_destroy(ucp_ep_h ep)
             ucp_worker_progress(worker);
             status = ucp_request_check_status(request);
         } while (status == UCS_INPROGRESS);
+
+        /* remove pending slow-path callback in case it was added in the above
+         * progress but won't be invoked */
+        ucs_callbackq_remove_if(&ep->worker->uct->progress_q,
```

Comment:
Not needed if do it at completions of `ucp_ep_close_nb` which is called from `ucp_disconnect_nb`

Reply:
ucp_ep_destroy calls progress which may insert the callback into the queue, so need to remove it if it's added (again).

Reply:
what do you mean 'over'? since both functions exist, both of them may be used.

Reply:
Right, but after ucp_ep_destroy calls ucp_ep_close_nb, it invokes progress. During this progress, an error may happen and the callback would be added to the queue. If it won't be invoked, need to remove it.

Reply:
Yes, this request will be completed when ucp_ep_close_nb is completed which does the same

Reply:
What if ucp_ep_close_nb is completed and then progress is called from ucp_ep_destroy and a new error happens?

Reply:
```
    request = ucp_disconnect_nb(ep);
    if (request == NULL) {
        goto out;

```
If ucp_ep_close_nb is completed progress won't be called